### PR TITLE
Document release process

### PR DESCRIPTION
**Pull request by jarrodmillman**

### CODE REVIEW

**Comment by jarrodmillman** [comment]
https://github.com/networkx/networkx/pull/5274

File: RELEASE.rst [hunk 1]
```diff
@@ -1,6 +1,22 @@
 Release process for ``numpydoc``
 ================================
 
+Introduction
+------------
+
+Example ``__version__``
+
+- 1.8rc1.dev0  # development version of 1.8 release candidate 1
```
Comment (rgommers) [inline RELEASE.rst:9]: These versions are slightly unusual, I'd expect:
```
1.2.0.dev0  # no rc1 in the name
1.2.0rc1
1.2.0
```

**Review by rgommers — COMMENTED** [review]
Good addition, testing against the largest users is the best way to smoke out potential issues. Overall this LGTM, just one minor comment.

File: RELEASE.rst (hunk 1 above)
Comment (jarrodmillman) [reply RELEASE.rst:9]: That works, but we need a version between `1.2.0rc1` and `1.2.0`.  So we could do:
```
1.2.0.dev0
1.2.0rc1
1.2.0rc2.dev0
1.2.0
```

Comment (rgommers) [reply RELEASE.rst:9]: Yes, that's how we do it in numpy and scipy as well

**Review by rgommers — APPROVED** [review]
Thanks Jarrod

**Pull request merged**